provisioning/transport/http/src/ProvisioningTransportHandlerHttp.cs

```diff
@@ -32,6 +32,20 @@ public ProvisioningTransportHandlerHttp()
             Proxy = DefaultWebProxySettings.Instance;
         }
 
+        /// <summary>
+        /// Registers a device described by the message.
+        /// </summary>
+        /// <param name="message">The provisioning message.</param>
+        /// <param name="timeout">The maximum amount of time to allow this operation to run for before timing out.</param>
+        /// <returns>The registration result.</returns>
+        public override async Task<DeviceRegistrationResult> RegisterAsync(
+            ProvisioningTransportRegisterMessage message,
+            TimeSpan timeout)
+        {
+            using var cts = new CancellationTokenSource(timeout);
+            return await RegisterAsync(message, cts.Token).ConfigureAwait(false);
```

Comment:
Since HTTP and MQTT do support using cancellation tokens all the way through, we can just create a cancellation token out of the timespan here.